bin/run.js

```diff
@@ -93,26 +107,37 @@ function main () {
   runTests(options)
 }
 
-function parseArgs (args) {
-  var options = {}
+function constructDefaultArgs () {
+  var defaultArgs = {
+    nodeArgs: [],
+    nycArgs: [],
+    timeout: process.env.TAP_TIMEOUT || 30,
+    color: supportsColor,
+    reporter: null,
+    files: [],
+    bail: false,
+    saveFile: null,
+    pipeToService: false,
+    coverageReport: null,
+  }
 
-  options.nodeArgs = []
-  options.nycArgs = []
-  options.timeout = process.env.TAP_TIMEOUT || 30
-  // coverage tools run slow.
-  /* istanbul ignore else */
   if (global.__coverage__) {
-    options.timeout = 240
+    defaultArgs.timeout = 240
   }
 
-  options.color = supportsColor
   if (process.env.TAP_COLORS !== undefined) {
-    options.color = !!(+process.env.TAP_COLORS)
+    defaultArgs.color = !!(+process.env.TAP_COLORS)
+  }
+
+  if (process.env.TAP === '1') {
+    defaultArgs.reporter = 'tap'
```

Comment:
This should actually be set on the final option object, not just the default set.